tests/unit_tests_Sampler_functionality.bash

```diff
@@ -212,6 +246,108 @@ function Clean_Tests_Environment_For_Following_Test__Sampler-validate-config-fil
     rm -r "${HYBRID_output_directory}"
 }
 
+function Make_Test_Preliminary_Operations__Sampler-validate-config-file-FIST()
+{
+    Make_Test_Preliminary_Operations__Sampler-create-input-file
+    HYBRID_module[Sampler]='FIST'
+    touch "${HYBRID_fist_module[Particle_file]}"
+    touch "${HYBRID_fist_module[Decays_file]}"
+}
+
+function Unit_Test__Sampler-validate-config-file-FIST()
+{
+    mkdir -p "${HYBRID_software_output_directory[Sampler]}"
+    cd "${HYBRID_software_output_directory[Sampler]}"
+    # Empty config file
+    touch "${HYBRID_software_configuration_file[Sampler]}"
+    Call_Codebase_Function_In_Subshell __static__Is_Sampler_Config_Valid &> /dev/null
+    if [[ $? -eq 0 ]]; then
+        Print_Error 'Config validation passed although config file is empty.'
+        return 1
+    fi
+    # Too many columns in config file
+    printf '%s\n' 'hypersurface whatever wrong' > "${HYBRID_software_configuration_file[Sampler]}"
+    Call_Codebase_Function_In_Subshell __static__Is_Sampler_Config_Valid &> /dev/null
+    if [[ $? -eq 0 ]]; then
+        Print_Error 'Config validation passed although config file has wrong number of columns.'
+        return 1
+    fi
+    # Repeated key in config file
+    printf '%s\n' 'output_file ~' 'output_file ~' > "${HYBRID_software_configuration_file[Sampler]}"
+    Call_Codebase_Function_In_Subshell __static__Is_Sampler_Config_Valid &> /dev/null
+    if [[ $? -eq 0 ]]; then
+        Print_Error 'Config validation passed although config file has repeated lines.'
+        return 1
+    fi
+    # Config file with invalid key
+    printf '%s\n' 'invalidKey value' > "${HYBRID_software_configuration_file[Sampler]}"
+    Call_Codebase_Function_In_Subshell __static__Is_Sampler_Config_Valid &> /dev/null
+    if [[ $? -eq 0 ]]; then
+        Print_Error 'Config validation passed although config file has invalid key.'
+        return 1
+    fi
+    # Config file missing required key 'hypersurface'
+    printf '%s\n' 'output_file .' > "${HYBRID_software_configuration_file[Sampler]}"
+    Call_Codebase_Function_In_Subshell __static__Is_Sampler_Config_Valid &> /dev/null
+    if [[ $? -eq 0 ]]; then
+        Print_Error 'Config validation passed although config file does not contain "hypersurface" key.'
+        return 1
+    fi
+    # Config file missing required key 'output_file'
+    printf '%s\n' "hypersurface $(which ls)" > "${HYBRID_software_configuration_file[Sampler]}"
+    Call_Codebase_Function_In_Subshell __static__Is_Sampler_Config_Valid &> /dev/null
+    if [[ $? -eq 0 ]]; then
+        Print_Error 'Config validation passed although config file does not contain "output_file" key.'
+        return 1
+    fi
```

Comment:
There are more than two required keys, isn't it? Here the tests are just a mere duplication of SMASH sampler, while for FIST something more is needed. I think that you should put 3 out of 4 mandatory keys and repeat for the 4 possible missing keys.